dask/base.py

```diff
@@ -1075,7 +1075,7 @@ def normalize_set(s):
     # Note: in some Python version / OS combinations, set order changes every
     # time you recreate the set (even within the same interpreter).
     # In most other cases, set ordering is consistent within the same interpreter.
-    return "set", _normalize_seq_func(sorted(s, key=str))
+    return "set", _normalize_seq_func(sorted(s, key=tokenize))
```

Comment:
This is quite expensive as it will tokenize everything twice. The benefit compared to str is only (1) when str is expensive and (2) when different objects have the same str output.